lib/state/ui/reducer.ts

```diff
@@ -22,4 +26,42 @@ const visiblePanes = (state = defaultVisiblePanes, { type, show }) => {
   return state;
 };
 
-export default combineReducers({ filteredNotes, visiblePanes });
+const setSelectedNote = (state = null, action) => {
+  switch (action.type) {
+    case 'App.notesLoaded': {
+      return state
+        ? action.notes.find(({ id }) => id === state.id) || state
+        : action.notes[0] || null;
+    }
+    case 'App.noteUpdatedRemotely':
+    case 'App.selectNote': {
+      return {
+        ...action.note,
+        hasRemoteUpdate: action.hasRemoteUpdate,
```

Comment:
did you check if this is used throughout the rest of the application?

if it is then we will want to add it to the type of `NoteEntity`. I'm guessing that we may not need it but in any case I'd rather have `hasRemoteUpdate?: boolean` on the type so we can rely on the type-checker to guard us